enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,171 @@
+This enhancement adds a new `Release` type to be used in `status` properties so we can distinguish between input and output schema.
+
+## Motivation
+
+Cincinnati provides [metadata about releases][cincinnati-metadata], including (since [here][cincinnati-channel-metadata]).
+Exposing that information to ClusterVersion consumers allows them to use live metadata, instead of talking to Cincinnati directly or hard-coding expected values.
+For example, the console currently [hard-codes an expected set of available channels][console-channels] for a given release; with this enhancement they could look up available channels in `status.desired.metadata["io.openshift.upgrades.graph.release.channels"]`.
+
+### Goals
+
+* Exposing release metadata provided by upstream Cincinnati for the currently-desired and available-update releases.
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+As proposed in [this API pull request][api-pull-request], to add a new type:
+
+
+```go
+// Release represents an OpenShift release in the release graph.
+// +k8s:deepcopy-gen=true
+type Release struct {
+  // version is a semantic versioning identifying the update version. When this
+  // field is part of spec, version is optional if image is specified.
+  Version string `json:"version"`
+
+  // image is a container image location that contains the update. When this
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // metadata is additional metadata associated with the release, such
+  // as associated channels (under
+  // io.openshift.upgrades.graph.release.channels) or a landing page
+  // (under url).
+  //
+  // +optional
+  Metadata map[string]string `json:"metadata,omitempty"`
+}
+```
+
+And to use that type instead of the current `Update` for
+`ClusterVersionStatus` properties.
+
+Then, in the cluster-version operator, to port existing logic like
+[available-update
+translation][cluster-version-operator-update-translation] and
+[available-update lookup][cluster-version-operator-update-lookup] to
+preserve the Cincinnati metadata.  In some cases (when an
```

Comment:
can we provide examples of implementation regarding:
1) if the user forces to a non-Cincinnati known release how will the status.desired look.
2) in cases if does know, how the status.desired will look.

Reply:
> can we provide examples of implementation regarding...

I've pushed fb0f876 -> 63745e6 adding this to the *Implementation Details/Notes/Constraints* section.